code/game/objects/items/weapons/RCD.dm

```diff
@@ -131,6 +131,61 @@
 	GLOB.rcd_list -= src
 	return ..()
 
+/obj/item/rcd/suicide_act(mob/living/user)
+	user.Immobilize(10 SECONDS) // You cannot move.
+	flags |= NODROP				// You cannot drop. You commit to die.
+	var/turf/suicide_tile = get_turf(src)
+	if(mode == MODE_DECON && checkResource(5, user))	// Same cost as deconstructing a wall.
+		user.visible_message("<span class='suicide'>[user] points [src] at [user.p_their()] chest and pulls the trigger. It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+		to_chat(user, "<span class='notice'>Deconstructing user...</span>")
+		var/obj/effect/temp_visual/rcd_effect/reverse/suicide_A = new(suicide_tile)
+		if(!do_after(user, 5 SECONDS))					// Deconstruction of most structures takes 5 seconds, wait for the animation to finish.
+			return
+		QDEL_NULL(suicide_A)
```

Comment:
Could you make the `do_after` 6,1 seconds?
And you should put the QDEL in the `if`-statement

Reply:
Moved.
Why 6,1 seconds (this just yeets after 1 second)?

Reply:
So the animation has enough time to play out

EDIT: the RCD deleted it aswell it seems, up to you how you want to do this

Reply:
> So the animation has enough time to play out
> 
> EDIT: the RCD deleted it aswell it seems, up to you how you want to do this

The visible portion of the animation is only 5 seconds long. The rest of it is invisible. I have visually compared the suicide with 5 and 6 seconds and 6 seconds creates a noticeable delay between the animation appearing to end and the dusting happening.

Reply:
As discussed in discord, fine with me